### DISCUSSION TITLE
Room Duration Time

### DISCUSSION THREAD

**Discussion by oksuzgil** [mynaparrot/plugNmeet-server#289] [General]
Hello, create room on api -> room duration parameter starts during creation. However, participants may not be connected to the room yet. The participant may even be tied up and waiting in the waiting room. For example, we will provide a 15-minute interview with a participant. It takes 3 minutes to create a room, distribute the links and connect those who will join the room. In this case, the interview will be only 12 minutes.

While the participant(s) are waiting in the waiting room, the time should start when the presenter connects and the first participant or participants approve the room. Even a delay time can be added with a second parameter. In other words, the room time starts +10s after the administrator connects and approves.

In addition, the 15-minute period has started, the interview continues and the interview needs to be extended. The administrator should be able to increase the room time to be added, such as +5min, +10min via the api.

**Comment by jibon57** [comment]
Hi,

The actual purpose of waiting room was to wait few moments before entering the session & a place from where admin can decide who can join or not. When you've entered to plugNmeet interface during that time you've started using extra CPU + bandwidth. In lot of enterprise they hire CPU for only specific time & there have involvement for extra cost. You application should have ability to prevent user to join the plugNmeet interface before start session. If you check our wordpress & moodle plugin then you'll see we've implemented this feature. Extending time by API will be little bit challenging but in future we'll have a look. 

Thanks
